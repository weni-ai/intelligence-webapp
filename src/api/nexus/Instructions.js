import nexusRequest from '../nexusaiRequest';
import { InstructionAdapter } from '../adapters/instructions/instruction';
import { useInstructionsStore } from '@/store/Instructions';

const request = nexusRequest;

export const Instructions = {
  async addInstruction({ projectUuid, instruction }) {
    const instructionsStore = useInstructionsStore();
    const body = {
      instructions: [...instructionsStore.instructions.data, instruction].map(
        InstructionAdapter.toApi,
      ),
    };
    const response = await request.$http.put(
      `api/${projectUuid}/customization/`,
      body,
      {
        hideGenericErrorAlert: true,
      },
    );

    const biggestId = response.data.instructions.reduce((acc, instruction) => {
      if (instruction.id > acc) acc = instruction.id;
      return acc;
    }, 0);

    return InstructionAdapter.fromApi(
      response.data.instructions.find(
        (instruction) => instruction.id === biggestId,
      ),
    );
  },

  async list({ projectUuid }) {
    const response = await request.$http.get(
      `api/${projectUuid}/customization/`,
    );

    return response.data.instructions.map(InstructionAdapter.fromApi);
  },

  async edit({ projectUuid, id, text }) {
    const instructionsStore = useInstructionsStore();

    const body = {
      instructions: [
        ...instructionsStore.instructions.data.filter(
          (instruction) => instruction.id !== id,
        ),
        { id, text },
      ].map(InstructionAdapter.toApi),
    };
    await request.$http.put(`api/${projectUuid}/customization/`, body);
  },

  async delete({ projectUuid, id }) {
    await request.$http.delete(`api/${projectUuid}/customization/?id=${id}`);
  },
};
