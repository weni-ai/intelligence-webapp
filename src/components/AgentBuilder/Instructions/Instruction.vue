<template>
  <section class="instruction">
    <p
      class="instruction__text"
      data-testid="instruction-text"
    >
      {{ text }}
    </p>

    <p
      v-if="tag"
      class="instruction__tag"
      data-testid="instruction-tag"
    >
      {{ tag }}
    </p>
  </section>
</template>

<script setup>
const props = defineProps({
  text: {
    type: String,
    required: true,
  },
  tag: {
    type: String,
    default: '',
    required: false,
  },
});
</script>

<style lang="scss" scoped>
.instruction {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: $unnnic-spacing-sm;

  padding: $unnnic-spacing-sm;

  &:not(:last-child) {
    border-bottom: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;
  }

  &__text {
    margin: 0;
    color: $unnnic-color-neutral-cloudy;

    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-gt;
    font-weight: $unnnic-font-weight-regular;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }

  &__tag {
    border-radius: $unnnic-border-radius-pill;

    padding: $unnnic-spacing-nano $unnnic-spacing-ant;

    background-color: $unnnic-color-neutral-light;
    color: $unnnic-color-neutral-darkest;
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-md;
    font-weight: $unnnic-font-weight-regular;
    line-height: $unnnic-font-size-body-md + $unnnic-line-height-md;
  }
}
</style>
