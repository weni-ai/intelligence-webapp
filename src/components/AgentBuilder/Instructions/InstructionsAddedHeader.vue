<template>
  <header class="instructions-added-header">
    <hgroup
      class="instructions-added-header__infos"
      data-testid="instructions-header-infos"
    >
      <h2
        class="infos__title"
        data-testid="instructions-title"
      >
        {{ $t('agent_builder.instructions.instructions_added.title') }}
      </h2>
      <h3
        class="infos__subtitle"
        data-testid="instructions-subtitle"
      >
        {{
          $t('agent_builder.instructions.instructions_added.subtitle', {
            count: instructionsCount,
          })
        }}
      </h3>
    </hgroup>

    <UnnnicSelectSmart
      v-model="filters"
      :options="filtersOptions"
      orderedByIndex
      multiple
      multipleWithoutSelectsMessage
      autocomplete
      data-testid="instructions-filters"
    />
  </header>
</template>

<script setup>
import { computed } from 'vue';
import i18n from '@/utils/plugins/i18n';

defineProps({
  instructionsCount: {
    type: Number,
    required: true,
  },
});

const filters = defineModel('filters', {
  type: Array,
  required: true,
});

const getFilterTranslation = (filter) =>
  i18n.global.t(
    `agent_builder.instructions.instructions_added.filters.${filter}`,
  );
const filtersOptions = computed(() => [
  {
    label: getFilterTranslation('placeholder'),
    value: '',
  },
  {
    label: getFilterTranslation('default_instructions'),
    value: 'default_instructions',
  },
  {
    label: getFilterTranslation('personalized_instructions'),
    value: 'personalized_instructions',
  },
]);
</script>

<style lang="scss" scoped>
.instructions-added-header {
  display: grid;
  grid-template-columns: 9fr 3fr;
  align-items: center;
  gap: $unnnic-spacing-sm;

  &__infos {
    display: flex;
    flex-direction: column;
    gap: $unnnic-spacing-nano;

    .infos__title {
      margin: 0;

      color: $unnnic-color-neutral-darkest;
      font-family: $unnnic-font-family-secondary;
      font-size: $unnnic-font-size-body-lg;
      font-weight: $unnnic-font-weight-bold;
      line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
    }

    .infos__subtitle {
      margin: 0;

      color: $unnnic-color-neutral-cloudy;
      font-family: $unnnic-font-family-secondary;
      font-size: $unnnic-font-size-body-md;
      font-weight: $unnnic-font-weight-regular;
      line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
    }
  }
}
</style>
