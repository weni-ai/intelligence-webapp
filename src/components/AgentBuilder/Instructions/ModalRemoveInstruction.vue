<template>
  <UnnnicModalDialog
    :modelValue="modelValue"
    data-testid="modal"
    class="modal-remove-instruction"
    showCloseIcon
    icon="warning"
    iconScheme="aux-red-500"
    :title="$t('agent_builder.instructions.remove_instruction.title')"
    size="sm"
    :secondaryButtonProps="{
      text: $t('agent_builder.instructions.remove_instruction.cancel'),
    }"
    :primaryButtonProps="{
      text: $t('agent_builder.instructions.remove_instruction.remove'),
      loading: instruction.status === 'loading',
      type: 'warning',
    }"
    @secondary-button-click="close"
    @primary-button-click="removeInstruction"
    @update:model-value="close"
  >
    <p
      class="modal-remove-instruction__description"
      data-testid="description"
    >
      {{
        $t('agent_builder.instructions.remove_instruction.modal_description')
      }}
    </p>
  </UnnnicModalDialog>
</template>

<script setup>
import { useInstructionsStore } from '@/store/Instructions';
import { ref } from 'vue';

const instructionsStore = useInstructionsStore();

const emit = defineEmits(['update:modelValue']);

const props = defineProps({
  instruction: {
    type: Object,
    required: true,
  },
});

const modelValue = ref(false);

function close() {
  emit('update:modelValue', false);
}

async function removeInstruction() {
  const { status } = await instructionsStore.removeInstruction(
    props.instruction.id,
  );

  if (status === 'complete') close();
}
</script>

<style lang="scss" scoped>
.modal-remove-instruction {
  :deep(.unnnic-modal-dialog__container__content) {
    padding-bottom: $unnnic-spacing-xs;
  }

  &__description {
    color: $unnnic-color-neutral-cloudy;

    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-gt;
    font-weight: $unnnic-font-weight-regular;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }
}
</style>
