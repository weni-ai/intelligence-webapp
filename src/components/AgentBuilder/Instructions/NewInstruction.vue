<template>
  <section
    class="new-instruction"
    data-testid="new-instruction"
  >
    <h2
      class="new-instruction__title"
      data-testid="new-instruction-title"
    >
      {{ $t('agent_builder.instructions.new_instruction.title') }}
    </h2>
    <UnnnicTextArea
      v-model="instructionsStore.newInstruction.text"
      data-testid="new-instruction-textarea"
      :placeholder="
        $t('agent_builder.instructions.new_instruction.textarea.placeholder')
      "
      :message="
        $t('agent_builder.instructions.new_instruction.textarea.description')
      "
    />
    <UnnnicButton
      class="new-instruction__add-instruction-button"
      data-testid="add-instruction-button"
      :disabled="!newInstruction.text.trim()"
      :text="$t('agent_builder.instructions.new_instruction.add_instruction')"
      :loading="newInstruction.status === 'loading'"
      @click="instructionsStore.addInstruction"
    />
  </section>
</template>

<script setup>
import { computed } from 'vue';
import { useInstructionsStore } from '@/store/Instructions';

const instructionsStore = useInstructionsStore();
const newInstruction = computed(() => instructionsStore.newInstruction);
</script>

<style lang="scss" scoped>
.new-instruction {
  border-radius: $unnnic-border-radius-md;
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;
  padding: $unnnic-spacing-sm;

  display: grid;
  grid-template-columns: repeat(12, 1fr);
  row-gap: $unnnic-spacing-sm;

  & > * {
    grid-column: 1 / -1;
  }

  &__title {
    margin: 0;

    color: $unnnic-color-neutral-darkest;
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-lg;
    font-weight: $unnnic-font-weight-bold;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }

  &__add-instruction-button {
    grid-column: 1 / 5;
  }
}
</style>
