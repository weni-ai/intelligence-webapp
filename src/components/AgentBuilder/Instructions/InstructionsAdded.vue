<template>
  <section class="instructions-added">
    <h2
      class="instructions-added__title"
      data-testid="instructions-title"
    >
      {{ $t('agent_builder.instructions.instructions_added.title') }}
    </h2>

    <section
      :class="[
        'instructions-added__instructions',
        {
          'instructions-added__instructions--loading':
            instructionsStore.instructions.status === 'loading',
        },
      ]"
      data-testid="instructions"
    >
      <template v-if="instructionsStore.instructions.status === 'loading'">
        <UnnnicSkeletonLoading
          v-for="index in 10"
          :key="index"
          tag="div"
          width="100%"
          height="60px"
          data-testid="instruction-loading"
        />
      </template>
      <template v-else>
        <Instruction
          v-for="(instruction, index) in instructionsDefault"
          :key="index"
          data-testid="instruction-default"
          :text="instruction"
          :tag="
            $t(
              'agent_builder.instructions.instructions_added.default_instruction',
            )
          "
        />
        <Instruction
          v-for="instruction in instructionsStore.instructions.data"
          :key="instruction.id"
          data-testid="instruction-added"
          :text="instruction.text"
        />
      </template>
    </section>
  </section>
</template>

<script setup>
import { computed } from 'vue';
import { useInstructionsStore } from '@/store/Instructions';

import Instruction from './Instruction.vue';
import i18n from '@/utils/plugins/i18n';

const instructionsStore = useInstructionsStore();
if (instructionsStore.instructions.status === null) {
  instructionsStore.loadInstructions();
}

const instructionsDefault = computed(() =>
  i18n.global.tm(
    'agent_builder.instructions.instructions_added.default_instructions',
  ),
);
</script>

<style lang="scss" scoped>
.instructions-added {
  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-sm;

  &__title {
    margin: 0;

    color: $unnnic-color-neutral-darkest;
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-lg;
    font-weight: $unnnic-font-weight-bold;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }

  &__instructions {
    display: flex;
    flex-direction: column;

    border-radius: $unnnic-border-radius-md;
    border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;

    &--loading {
      gap: $unnnic-spacing-nano;

      border: none;
      border-radius: 0;
    }
  }
}
</style>
