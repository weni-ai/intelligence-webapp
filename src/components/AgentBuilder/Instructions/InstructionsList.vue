<template>
  <section
    :class="[
      'instructions-list',
      {
        'instructions-list--loading': isLoading,
      },
    ]"
    data-testid="instructions"
  >
    <template v-if="isLoading">
      <UnnnicSkeletonLoading
        v-for="index in 10"
        :key="index"
        tag="div"
        width="100%"
        height="60px"
        data-testid="instruction-loading"
      />
    </template>
    <template v-else>
      <Instruction
        v-for="instruction in instructions"
        :key="instruction.id"
        :instruction="instruction"
        :showActions="showActions"
        data-testid="instruction-added"
      />
    </template>
  </section>
</template>

<script setup>
import Instruction from './Instruction.vue';

defineProps({
  instructions: {
    type: Array,
    required: true,
  },
  isLoading: {
    type: Boolean,
    default: false,
  },
  showActions: {
    type: Boolean,
    default: false,
  },
});
</script>

<style lang="scss" scoped>
.instructions-list {
  display: flex;
  flex-direction: column;

  border-radius: $unnnic-border-radius-md;
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;

  &--loading {
    gap: $unnnic-spacing-nano;

    border: none;
    border-radius: 0;
  }
}
</style>
