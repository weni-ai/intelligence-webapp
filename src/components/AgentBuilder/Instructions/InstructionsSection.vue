<template>
  <section class="instructions-section">
    <h2
      class="instructions-section__title"
      data-testid="instructions-section-title"
    >
      {{ $t('agent_builder.instructions.instructions_list.title') }}
    </h2>

    <UnnnicTab
      :tabs="tabs"
      :activeTab="activeTab"
      data-testid="instructions-section-tab"
      @change="instructionsStore.activeInstructionsListTab = $event"
    >
      <template
        v-for="tab in tabs"
        :key="tab"
        #[`tab-head-${tab}`]
      >
        {{ $t(`agent_builder.instructions.instructions_list.tabs.${tab}`) }}
      </template>
      <template #[`tab-panel-${activeTab}`]>
        <p class="instructions-section__tab-description">
          {{
            $t(
              `agent_builder.instructions.instructions_list.tabs.${activeTab}_description`,
            )
          }}
        </p>
      </template>
    </UnnnicTab>
    <InstructionsList
      v-if="activeTab === 'custom'"
      data-testid="instructions-custom"
      :instructions="instructionsStore.instructions.data"
      :isLoading="instructionsStore.instructions.status === 'loading'"
      showActions
    />
    <InstructionsList
      v-if="activeTab === 'default'"
      data-testid="instructions-default"
      :instructions="instructionsDefault"
      :isLoading="false"
      :showActions="false"
    />
    <InstructionsList
      v-if="activeTab === 'safety_topics'"
      data-testid="instructions-safety-topics"
      :instructions="instructionsSafetyTopics"
      :isLoading="false"
      :showActions="false"
    />
  </section>
</template>

<script setup>
import { computed, ref } from 'vue';
import { useInstructionsStore } from '@/store/Instructions';

import InstructionsList from './InstructionsList.vue';
import i18n from '@/utils/plugins/i18n';

const instructionsStore = useInstructionsStore();

if (instructionsStore.instructions.status === null) {
  instructionsStore.loadInstructions();
}

const tabs = ref(['custom', 'default', 'safety_topics']);
const activeTab = computed(() => instructionsStore.activeInstructionsListTab);

function generateMockedInstructions(type) {
  return i18n.global
    .tm(`agent_builder.instructions.instructions_list.${type}`)
    .map((instruction, index) => ({
      id: `${type}-${index}`,
      text: instruction,
    }));
}

const instructionsDefault = computed(() =>
  generateMockedInstructions('default_instructions'),
);

const instructionsSafetyTopics = computed(() =>
  generateMockedInstructions('safety_topics'),
);
</script>

<style lang="scss" scoped>
.instructions-section {
  display: flex;
  flex-direction: column;

  &__title {
    margin: 0 0 $unnnic-spacing-sm;

    color: $unnnic-color-neutral-darkest;
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-lg;
    font-weight: $unnnic-font-weight-bold;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }

  &__tab-description {
    margin-bottom: $unnnic-spacing-xs;

    color: $unnnic-color-neutral-cloudy;

    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-gt;
    font-weight: $unnnic-font-weight-regular;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }
}
</style>
