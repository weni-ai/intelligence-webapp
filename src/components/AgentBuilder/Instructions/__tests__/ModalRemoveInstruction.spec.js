import { mount } from '@vue/test-utils';
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import { createTestingPinia } from '@pinia/testing';

import ModalRemoveInstruction from '../ModalRemoveInstruction.vue';
import i18n from '@/utils/plugins/i18n';
import { useInstructionsStore } from '@/store/Instructions';

describe('ModalRemoveInstruction.vue', () => {
  let wrapper;
  let instructionsStore;

  const pinia = createTestingPinia({
    initialState: {
      Instructions: {
        instructions: {
          data: [],
        },
      },
    },
  });

  const mockInstruction = {
    id: 'test-instruction-id',
    text: 'Test instruction text',
    status: null,
  };

  const SELECTORS = {
    modal: '[data-testid="modal"]',
    description: '[data-testid="description"]',
  };
  const find = (selector) => wrapper.find(SELECTORS[selector]);
  const findComponent = (component) =>
    wrapper.findComponent(SELECTORS[component]);

  beforeEach(() => {
    instructionsStore = useInstructionsStore(pinia);
    instructionsStore.removeInstruction = vi.fn().mockResolvedValue({
      status: 'complete',
    });

    wrapper = mount(ModalRemoveInstruction, {
      props: {
        modelValue: true,
        instruction: mockInstruction,
      },
      global: {
        plugins: [pinia],
      },
    });
  });

  afterEach(() => {
    wrapper.unmount();
    vi.clearAllMocks();
  });

  describe('Component rendering', () => {
    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders the modal with correct props', () => {
      const modal = findComponent('modal');
      expect(modal.exists()).toBe(true);
      expect(modal.props('modelValue')).toBe(true);
      expect(modal.props('title')).toBe(
        i18n.global.t('agent_builder.instructions.remove_instruction.title'),
      );
      expect(modal.props('showCloseIcon')).toBe(true);
      expect(modal.props('icon')).toBe('warning');
      expect(modal.props('iconScheme')).toBe('aux-red-500');
      expect(modal.props('size')).toBe('sm');
    });

    it('renders the correct description', () => {
      expect(find('description').exists()).toBe(true);
      expect(find('description').text()).toBe(
        i18n.global.t(
          'agent_builder.instructions.remove_instruction.modal_description',
        ),
      );
    });

    it('renders the correct button props', () => {
      const modal = findComponent('modal');
      expect(modal.props('secondaryButtonProps')).toEqual({
        text: i18n.global.t(
          'agent_builder.instructions.remove_instruction.cancel',
        ),
      });

      expect(modal.props('primaryButtonProps')).toEqual({
        text: i18n.global.t(
          'agent_builder.instructions.remove_instruction.remove',
        ),
        loading: false,
        type: 'warning',
      });
    });
  });

  describe('Modal interactions', () => {
    it('emits update:modelValue when modal is closed via close icon', async () => {
      await findComponent('modal').vm.$emit('update:model-value', false);

      expect(wrapper.emitted('update:modelValue')).toBeTruthy();
      expect(wrapper.emitted('update:modelValue')[0]).toEqual([false]);
    });

    it('emits update:modelValue when secondary button is clicked', async () => {
      await findComponent('modal').vm.$emit('secondary-button-click');

      expect(wrapper.emitted('update:modelValue')).toBeTruthy();
      expect(wrapper.emitted('update:modelValue')[0]).toEqual([false]);
    });
  });

  describe('Remove instruction functionality', () => {
    it('calls removeInstruction with correct instruction id', async () => {
      await findComponent('modal').vm.$emit('primary-button-click');

      expect(instructionsStore.removeInstruction).toHaveBeenCalledWith(
        mockInstruction.id,
      );
    });

    it('closes modal when removal is successful', async () => {
      await findComponent('modal').vm.$emit('primary-button-click');
      await wrapper.vm.$nextTick();

      expect(wrapper.emitted('update:modelValue')).toBeTruthy();
      expect(wrapper.emitted('update:modelValue')[0]).toEqual([false]);
    });

    it('does not close modal when removal fails', async () => {
      instructionsStore.removeInstruction.mockResolvedValue({
        status: 'error',
      });

      await findComponent('modal').vm.$emit('primary-button-click');
      await wrapper.vm.$nextTick();

      expect(wrapper.emitted('update:modelValue')).toBeFalsy();
    });
  });

  describe('Loading states', () => {
    it('shows loading state on primary button when instruction status is loading', async () => {
      await wrapper.setProps({
        instruction: { ...mockInstruction, status: 'loading' },
      });

      expect(findComponent('modal').props('primaryButtonProps').loading).toBe(
        true,
      );
    });

    it('does not show loading state when instruction status is not loading', () => {
      expect(findComponent('modal').props('primaryButtonProps').loading).toBe(
        false,
      );
    });
  });
});
