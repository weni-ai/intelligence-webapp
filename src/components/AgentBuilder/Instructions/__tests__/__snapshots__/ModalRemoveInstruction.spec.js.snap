// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModalRemoveInstruction.vue > Component rendering > should match snapshot 1`] = `
"<section data-v-ab1febba="" data-v-18127a74="" class="unnnic-modal-dialog modal-remove-instruction" data-testid="modal">
  <section data-v-ab1febba="" class="unnnic-modal-dialog__overlay" data-testid="modal-overlay"></section>
  <section data-v-ab1febba="" class="unnnic-modal-dialog__container unnnic-modal-dialog__container--sm" data-testid="modal-container">
    <!---->
    <section data-v-ab1febba="" class="unnnic-modal-dialog__container__body">
      <header data-v-ab1febba="" class="unnnic-modal-dialog__container__header">
        <section data-v-ab1febba="" class="unnnic-modal-dialog__container__title-container"><span data-v-9846e6db="" data-v-ab1febba="" class="material-symbols-rounded unnnic-icon-scheme--aux-red-500 unnnic-icon-size--md unnnic-modal-dialog__container__title-icon" data-testid="title-icon" translate="no">warning</span>
          <h1 data-v-ab1febba="" class="unnnic-modal-dialog__container__title-text" data-testid="title-text">Remove instruction</h1>
        </section><span data-v-9846e6db="" data-v-ab1febba="" class="material-symbols-rounded unnnic-icon-scheme--neutral-darkest unnnic-icon-size--md unnnic--clickable" data-testid="close-icon" translate="no">close</span>
      </header>
      <section data-v-ab1febba="" class="unnnic-modal-dialog__container__content">
        <p data-v-18127a74="" class="modal-remove-instruction__description" data-testid="description">Are you sure you want to remove the instruction?</p>
      </section>
      <section data-v-ab1febba="" data-testid="actions-section" class="unnnic-modal-dialog__container__actions"><button data-v-d127031e="" data-v-ab1febba="" data-testid="secondary-button" class="unnnic-modal-dialog__container__actions__secondary-button unnnic-button unnnic-button--size-large unnnic-button--tertiary unnnic-modal-dialog__container__actions__secondary-button">
          <!---->
          <!---->
          <!----><span data-v-d127031e="" class="unnnic-button__label" data-testid="button-label"> Cancel</span>
          <!---->
        </button><button data-v-d127031e="" data-v-ab1febba="" data-testid="primary-button" class="unnnic-modal-dialog__container__actions__primary-button unnnic-button unnnic-button--size-large unnnic-button--warning unnnic-modal-dialog__container__actions__primary-button">
          <!---->
          <!---->
          <!----><span data-v-d127031e="" class="unnnic-button__label" data-testid="button-label"> Remove</span>
          <!---->
        </button></section>
    </section>
  </section>
</section>"
`;
